### ISSUE TITLE
build: Issues with build on SunOS/Illumos

### ISSUE THREAD

**Issue by rohit-saharan** [minio/minio#3586]
Hi, I've tried compiling this on SunOS ( illumos ) with Go v1.7. Getting same with using `make` after modifying it a bit to run.

Here's the error : 
```
go build
# github.com/minio/minio/cmd
cmd/config-migrate.go:101: undefined: removeAll
cmd/posix.go:132: undefined: mkdirAll
cmd/posix.go:146: undefined: disk.GetInfo
cmd/posix.go:217: undefined: isValidVolname
cmd/posix.go:313: undefined: isValidVolname
cmd/posix.go:628: undefined: mkdirAll
cmd/posix.go:938: undefined: mkdirAll
```

**Comment by harshavardhana** [comment]
Minio is not supported on sunos for now.

**Comment by rohit-saharan** [comment]
I will try to help if possible.

**Comment by harshavardhana** [comment]
> I will try to help if possible.

Thanks a @geekrohit - feel free to send us a PR.

**Comment by rohit-saharan** [comment]
Ok, on master , while trying to build, am getting this error : 
```
# github.com/minio/minio/pkg/lock
pkg/lock/lock.go:83: undefined: LockedOpenFile
```
Any idea ? I've tried looking for solutions, but cannot find the LockedOpenFile in `os` or `sync` packages.
The other errors that I had posted above are not visible now.

**Comment by harshavardhana** [comment]
> Any idea ? I've tried looking for solutions, but cannot find the LockedOpenFile in os or sync packages.
> The other errors that I had posted above are not visible now.

Yes we added a new code which doesn't build on SunOS

**Comment by rohit-saharan** [comment]
Ok, so any alternatives I can use in that place ?
You're already supporting freebsd, so maybe we can get this running on illumos as well.

**Comment by harshavardhana** [comment]
> You're already supporting freebsd, so maybe we can get this running on illumos as well.

FreeBSD because we constantly test against this operating system and there is an active ZFS community.   We never tested our code on illumos so we can't guarantee the stability of Minio on such a platform, neither do we have such an infrastructure in-place to validate. 

To the fix the build issue for illumos you have to work your way up from lower most optimizations which we did for each platforms. ` git grep 'build linux'` would help.

**Comment by rohit-saharan** [comment]
@harshavardhana Ok, illumos has an active community for ZFS as well, and part of OpenZFS alliance. If you want to test on illumos and want a build machine, we can do that, let me know.

Till then I'll go through your suggestion.

**Comment by harshavardhana** [comment]
@geekrohit do you need any help from us - if you give us access to the setup we can do this for you.

**Comment by harshavardhana** [comment]
Fixed by #3719
